packages/jtex/src/cli/check.ts

```diff
@@ -0,0 +1,240 @@
+import { Command } from 'commander';
+import yaml from 'js-yaml';
+import { clirun } from './utils';
+import fs from 'fs';
+import { join } from 'path';
+import type { ISession } from '../types';
+import { REDERER_DOC_KEYS, DOC_FRONTMATTER_KEYS } from '../types';
+import { validateTemplateConfig } from '../validators';
+import type { ValidationOptions } from '@curvenote/validators';
+import chalk from 'chalk';
+
+type Variables = Record<'options' | 'doc' | 'parts' | 'packages', Record<string, number[]>>;
+
+function matchOpts(line: string, lineNumber: number, variables: Variables) {
+  const match = line.match(/\[[#|-]-?([^\]#]*)-?[#|-]\]/g);
+  if (!match) return;
+  match.forEach((template) => {
+    const args = template.match(/(doc|options|tagged|parts)\.([a-zA-Z0-9_]+)/g);
+    if (!args) return;
+    args.forEach((a) => {
+      const [kind, value] = a.split('.');
+      const useKind = (kind === 'tagged' ? 'parts' : kind) as keyof typeof variables;
+      if (!variables[useKind]?.[value]) variables[useKind][value] = [];
+      if (!variables[useKind][value].includes(lineNumber)) {
+        variables[useKind][value].push(lineNumber);
+      }
+    });
+  });
+}
+
+function lineNumbersToString(lineNumbers: number[]): string {
+  return `line${lineNumbers.length > 1 ? 's' : ''} ${lineNumbers.join(', ')}`;
+}
+
+function extractVariablesFromTemplate(template: string) {
+  const variables: Variables = {
+    options: {},
+    doc: {},
+    parts: {},
+    packages: {},
+  };
+  template.split('\n').forEach((line: string, i: number) => {
+    matchOpts(line, i + 1, variables);
+  });
+  template
+    .split('\n')
+    .map((line) => line.split(/(?<!\\)%/)[0]) // Strip comments (backwards lookup for \% escapes)
+    .map((line: string, i) => {
+      const match = line.match(
+        /\\(?:usepackage|RequirePackage)(?:\[(?:[^\]]+)\])?\{([a-zA-z0-9_-]+)\}/g,
+      );
+      if (!match) return [];
+      return { match, lineNumber: i + 1 };
+    })
+    .flat()
+    .filter((m) => !!m)
+    .forEach((p) => {
+      const name = /\{([a-zA-z0-9_-]+)\}/.exec(p.match[0])?.[1];
+      if (!name) return;
+      if (!variables.packages?.[name]) variables.packages[name] = [];
+      if (!variables.packages[name].includes(p.lineNumber)) {
+        variables.packages[name].push(p.lineNumber);
+      }
+    });
+  return variables;
+}
+
+function printWarnings(
+  session: ISession,
+  name: string,
+  messages: Required<ValidationOptions['messages']>,
+) {
+  if (messages.errors.length === 0 && messages.warnings.length === 0) return false;
+  session.log.info(`${name}`);
+
+  messages.warnings?.forEach((e) => {
+    session.log.warn(`  [${e.property}] ${e.message}`);
+  });
+  messages.errors?.forEach((e) => {
+    session.log.error(`  [${e.property}] ${e.message}`);
+  });
+  messages.warnings = [];
+  messages.errors = [];
+  return true;
+}
+
+export function checkTemplate(session: ISession, path: string) {
```

Comment:
Wondering if this should also check that `[-CONTENT-]` is present?

Reply:
Yep, and imports, good call.

Reply:
![image](https://user-images.githubusercontent.com/913249/189970303-3557170b-8609-4b6f-9f2f-49b4eb3a3325.png)